### ISSUE TITLE
Specify the number of decimal places

### ISSUE THREAD

**Issue by funlennysub** [neosmart/prettysize-rs#12]
From what I've read in the source code, there's no way to set the number of decimal places. I'm making an app where it kind of matters, because when I have `Size::from_bytes(170_516_114)` which should be around 162.6 MiB, but lib shows it as 163 MiB and in other cases (examples from the docs etc seems to display fine) it just seems to round up the number which is not ideal but not a top priority for me right now.

Would it be something that could be added in 0.5 version? From what I saw, there's just a map of rules on how to format, so I'm not sure how hard would it be to add this feature.

**Comment by mqudsi** [comment]
This wouldn't be part of the default formatting rules where the heuristic is "lower precision as magnitude increases", but it would be added to the `SizeFormatter` object (new `precision` field and new `with_precision()` builder function).

**Comment by funlennysub** [comment]
Okay, I'll wait for the update then. Thanks.

**Comment by mqudsi** [comment]
I'm happy to work on this but I'll likely push it to 0.6 as I think we have a good 0.5 release mostly ready.

**Comment by mqudsi** [comment]
This didn't take as long as I expected. Can you try it out and let me know how it works for you?

Do this in `Cargo.toml`:

```toml
[dependencies]
size = { git = "https://github.com/neosmart/prettysize-rs.git", branch = "0.6.x" }
```

You can control the scale from the `Size` object directly, passing in `size.format().with_scale(Some(3))` to `format!()` or `write!()` or whatever, or use a `SizeFormatter` with the `with_scale(...)` option set:

![image](https://github.com/neosmart/prettysize-rs/assets/606923/81dff3f2-c911-4aa0-958a-b557acede604)

**Comment by funlennysub** [comment]
Seems to work as expected. Appreciate the quick work!
![image](https://github.com/neosmart/prettysize-rs/assets/26184007/a184aeb0-ef87-4163-8f5b-b569064b12b6)

**Comment by mqudsi** [comment]
Great! If any issues arise or you have any feedback on the api, please feel free to open another issue or comment here. I'm closing this as it appears to be addressed now.

**Comment by mqudsi** [comment]
As prettysize-rs 0.5.0 has been released, I've merged the 0.6.x branch into master which now contains the dynamic scaling code.
